gprofiler/profilers/java.py

```diff
@@ -273,11 +302,91 @@ def make_application_metadata(self, process: Process) -> Dict[str, Any]:
         # that loads other libs.
         libjvm_elfid = get_mapped_dso_elf_id(process, "/libjvm")
 
-        metadata = {"java_version": version, "libjvm_elfid": libjvm_elfid}
+        try:
+            jvm_flags = self.get_jvm_flags_serialized(process)
+        except CalledProcessError as e:
+            jvm_flags = [{"error": e.stderr}]
+        except Exception as e:
+            logger.exception("Failed to collect JVM flags", pid=process.pid)
+            jvm_flags = [{"error": type(e).__name__}]
+
+        metadata = {
+            "java_version": version,
+            "libjvm_elfid": libjvm_elfid,
+            "jvm_flags": jvm_flags,
+        }
 
         metadata.update(super().make_application_metadata(process))
         return metadata
 
+    def get_jvm_flags_serialized(self, process: Process) -> Optional[List[Dict]]:
+        jvm_flags = self.get_jvm_flags(process)
+        if jvm_flags is None:
+            return None
+
+        return [flag.to_dict() for flag in sorted(jvm_flags, key=lambda flag: flag.name)]
+
+    def get_jvm_flags(self, process: Process) -> Optional[Iterable[JvmFlag]]:
+        if self.java_collect_jvm_flags == JavaFlagCollectionOptions.NONE:
+            return None
+
+        jvm_raw_flags: Iterable[JvmFlag] = filter(self.filter_jvm_flag, self.get_jvm_flags_raw(process))
```

Comment:
Don't name it `raw` flags like the function `_raw` because it did go through filtering.

Function can be `_unfiltered`

Variable can be `_basic_filters` or sth like that